webapp/frontPage/js/modal.js

```diff
@@ -395,14 +402,25 @@ define(['jquery', 'magnific', 'util'], function($, magnific, util) {
             'X-LABKEY-CSRF': LABKEY.CSRF
           }
         }).success(function() {
-            window.location = LABKEY.ActionURL.buildURL("cds", "app.view?"); // set password should log user in automatically
-        }).error(function(e) {
-          var errorMsg = 'Create account failed. ';
-          if (e && e.responseJSON && e.responseJSON.errors && e.responseJSON.errors.length > 0) {
-            errorMsg = errorMsg + e.responseJSON.errors[0].message;
-          }
-          $('.create-account-modal .notifications p').html(errorMsg);
-        });
+
+            //sets NeedSurvey to true
+            $.ajax({
+              url: LABKEY.ActionURL.buildURL("cds", "updateNeedSurvey.api"),
+              method: 'POST',
+              data: {
+                'X-LABKEY-CSRF': LABKEY.CSRF
+              }
+            }).success(function() {
+              window.location = LABKEY.ActionURL.buildURL("cds", "app.view", null, {NeedSurvey:true}); // set password should log user in automatically
```

Comment:
instead of setting NeedSurvey to true, which would trigger another redirect in controller, could instead use param "survey=true", which is actually what prompts survey page.

Reply:
done!